public/js/acmeair-common.js

```diff
@@ -38,77 +38,63 @@ function hideWaitDialog() {
 }
 
 function updateLoggedInUserWelcome() {
-	var loggedinuser = dojo.cookie("loggedinuser");
-	if (loggedinuser == null) {
+	const session = dojo.cookie("acmeair-user");
+
+	if (!session) {
 		dojo.byId("loggedinwelcome").innerHTML = '';
-	}
-	else {
-		dojo.byId("loggedinwelcome").innerHTML = 'Welcome Back ' + loggedinuser;
+	} else {
+		const data = session.split(';')
+		dojo.byId("loggedinwelcome").innerHTML = 'Welcome Back ' + data[2];
 	}
 }
 
 function login() {
 	hideLoginDialog();
 	showLoginWaitDialog();
 	
-	var userString = document.getElementById('userId').value;
 	dojo.xhrPost({
 		content : {
-			login: userString,
+			login: document.getElementById('userId').value,
 			password: document.getElementById('password').value
 		},
 		url: 'rest/api/login',
-		load: function(response, ioArgs) {
-			hideLoginWaitDialog();
-			if (response != 'logged in') {
-				// TODO: why isn't error function being called in this case
-				alert('error logging in, response: ' + response);
-				return;
+		handleAs: 'json',
+		load: (response) => { 
+			if (!response) {
+				throw new Error('trouble logging in')
 			}
-			dojo.cookie("loggedinuser", userString, {expires: 5});
+
+			hideLoginWaitDialog();
+			const sessionData = `${response.sessionId};${response.customerId};${response.email}`
```

Comment:
Why using `;` as a separator ? Maybe using a JSON would be easier (and avoid the `data[2]` in a previous comment). I believe a cookie should be able to store all this data without issue

Reply:
I was trying to keep the size of the cookie down.  JSON adds overhead to the size with all the extra characters.